docs: samples: subsys: lorawan: class_a: clarify LoRaWAN example README limitations and guidance

felf-zhaw opened the pull request:
### Summary
This pull request updates the README for the LoRaWAN Class A example to clarify limitations for multiple runs and provide guidance for resolving the issue. It addresses the problem of a hardcoded `dev_nonce` causing join failures.

### Changes Made
- Added a new "Important Notes for Multiple Runs" section in the README.
- Explained the limitation with the `dev_nonce`.
- Provided solutions, including enabling `CONFIG_LORAWAN_NVM_SETTINGS` or programmatically setting a random `dev_nonce`.

Fixes #83920

JordanYates reviewed (changes requested):
IMO the notes needs to follow this form:

1. Example only works once by default, because `dev_nonce` MUST increment on each join request.
2. To resolve the above, the `dev_nonce` must be incremented on each request, and preserved across reboots.
3. Zephyr provides a builtin implementation of the above with `CONFIG_LORAWAN_NVM_SETTINGS

JordanYates commented on samples/subsys/lorawan/class_a/README.rst:
```diff
@@ -25,3 +25,16 @@ The following commands build and flash the sample.
+Important Notes for Multiple Runs
+*********************************
+
+By default, this example works only once with the current configuration. On subsequent attempts, the device may fail to join the LoRaWAN network due to a hardcoded ``dev_nonce`` value. According to the LoRaWAN specification, ``dev_nonce`` must increment for every new connection attempt. To ensure the sample works on multiple runs, follow these recommendations:
```
By default, this example will once succeed the first time it is run.

JordanYates replied:
On subsequent attempts, the LoRaWAN network server may reject the join request due to ...

JordanYates replied:
To run this sample multiple times, choose one of the following options:

JordanYates commented on samples/subsys/lorawan/class_a/README.rst:
```diff
@@ -25,3 +25,16 @@ The following commands build and flash the sample.
+1. **Manually Increment ``dev_nonce``:**
+   Modify the sample code to increment ``join_cfg.otaa.dev_nonce`` before each connection attempt and ensure it is preserved across reboots.
+
+2. **Zephyr Built-in Implementation:**
```
** Built-in Zephyr Settings Implementation:**

JordanYates reviewed (changes requested):
See comments

felf-zhaw committed:
docs: samples: subsys: lorawan: class_a: update README

Updated the README for the LoRaWAN Class A example to explain limitations
when running multiple times. Provided guidance to resolve the issue.

Signed-off-by: Flavio Felder <felf@zhaw.ch>

kartben reviewed (commented):
thank you very much @felf-zhaw for this very useful PR!

kartben reviewed (approved):
thank you very much @felf-zhaw for this very useful PR!
